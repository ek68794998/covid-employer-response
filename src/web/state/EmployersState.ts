import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}