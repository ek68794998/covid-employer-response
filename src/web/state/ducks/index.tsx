import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});