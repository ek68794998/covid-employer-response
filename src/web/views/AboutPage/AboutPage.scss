@import "../Dimensions.scss";

main#about {
	margin: 0 auto;
	width: 100%;

	.AboutPage__Content {
		line-height: 180%;
		margin: 0 auto;
		max-width: 1440px;
		padding: $page-padding;

		.AboutPage__CitationTypes ul {
			line-height: 220%;
			list-style: none;
			position: relative;

			li::before {
				font-family: 'Material Icons';
				font-size: 18px;
				left: 16px;
				position: absolute;
			}

			li:nth-child(1)::before {
				content: "description";
			}

			li:nth-child(2)::before {
				content: "how_to_reg";
			}

			li:nth-child(3)::before {
				content: "hearing";
			}
		}
	}

	h2 {
		margin-top: 32px;

		&:first-child {
			margin-top: 0;
		}
	}

	.material-icons {
		user-select: none;
	}
}

@media (max-width: $body-narrow-width) {
	main#about .AboutPage__Content {
		padding: $page-padding-narrow;
	}
}
