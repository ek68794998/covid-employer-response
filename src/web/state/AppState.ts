import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}