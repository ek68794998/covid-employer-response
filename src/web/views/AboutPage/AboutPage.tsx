import React from "react";
import { Helmet } from "react-helmet";
import ReactMarkdown from "react-markdown";
import { useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

import { format, LocalizedStrings } from "../../../common/LocalizedStrings";
import { ProjectUrl } from "../../../common/constants/UrlConstants";

import { getStrings } from "../../state/ducks/localization/selectors";

import "./AboutPage.scss";

const AboutPage: React.FC = (): React.ReactElement => {
	const strings: LocalizedStrings = useSelector(getStrings);

	const coronavirusWikipediaUrl: string = "https://en.wikipedia.org/wiki/Coronavirus_disease_2019";
	const coronavirusPandemicWikipediaUrl: string = "https://en.wikipedia.org/wiki/2019%E2%80%9320_coronavirus_pandemic";

	return (
		<main id="about">
			<Helmet>
				<title>{strings.about} | {strings.appTitle}</title>
			</Helmet>
			<div className="AboutPage__Content">
				<h2 id="what-is">{format(strings.aboutSectionHeaders.whatIs, { app: strings.appTitleShort })}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={(
						format(
							strings.aboutSectionParagraphs.whatIs,
							{
								app: strings.appTitleShort,
								appFullName: strings.appTitle,
								covidWikiUrl: coronavirusWikipediaUrl,
								githubUrl: ProjectUrl,
								pandemicWikiUrl: coronavirusPandemicWikipediaUrl,
							})
					)}
				/>

				<h2 id="contributing">{strings.aboutSectionHeaders.contributing}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={(
						format(
							strings.aboutSectionParagraphs.contributing,
							{
								app: strings.appTitleShort,
								githubUrl: ProjectUrl,
							})
					)}
				/>

				<h2 id="reports-and-claims">{strings.aboutSectionHeaders.reportsAndClaims}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={strings.aboutSectionParagraphs.reportsAndClaims}
				/>

				<h2 id="claim-processing">{strings.aboutSectionHeaders.claimProcessing}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={strings.aboutSectionParagraphs.claimProcessing}
				/>

				<h2 id="submit-claims">{strings.aboutSectionHeaders.submitClaims}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={(
						format(
							strings.aboutSectionParagraphs.submitClaims,
							{
								submit: strings.submit,
							})
					)}
				/>

				<h2 id="citation-types">{strings.aboutSectionHeaders.citationTypes}</h2>
				<ReactMarkdown
					className="AboutPage__CitationTypes"
					linkTarget="_blank"
					source={(
						format(
							strings.aboutSectionParagraphs.citationTypes,
							{
								hearsay: strings.citationTypes.hearsay,
								hearsayDescription: strings.citationTypeDescriptions.hearsay,
								publication: `${strings.citationTypes.publication}`,
								publicationDescription: strings.citationTypeDescriptions.publication,
								statement: strings.citationTypes.statement,
								statementDescription: strings.citationTypeDescriptions.statement,
							})
					)}
				/>

				<h2 id="employer-ratings">{strings.aboutSectionHeaders.employerRatings}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={strings.aboutSectionParagraphs.employerRatings}
				/>

				<h2 id="layoffs-and-furloughs">{strings.aboutSectionHeaders.layoffsAndFurloughs}</h2>
				<ReactMarkdown
					linkTarget="_blank"
					source={strings.aboutSectionParagraphs.layoffsAndFurloughs}
				/>
			</div>
		</main>
	);
};

export default withRouter(AboutPage);
