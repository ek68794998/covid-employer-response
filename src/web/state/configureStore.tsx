import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;