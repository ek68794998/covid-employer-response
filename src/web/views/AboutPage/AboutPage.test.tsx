import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { ploc } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import AboutPage from "./AboutPage";

describe("<AboutPage />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({
			strings: {
				about: ploc("about"),
				aboutSectionHeaders: {
					citationTypes: `${ploc("citationTypesHeader")}: {publication} / {statement} / {hearsay}`,
					claimProcessing: ploc("claimProcessingHeader"),
					contributing: ploc("contributingHeader"),
					employerRatings: ploc("employerRatingsHeader"),
					reportsAndClaims: ploc("reportsAndClaimsHeader"),
					submitClaims: ploc("submitClaimsHeader"),
					whatIs: `${ploc("whatIsHeader")}: {app}`,
				},
				appTitle: ploc("appTitle"),
				appTitleShort: ploc("app"),
				citationTypeDescriptions: {
					hearsay: ploc("hearsayDescription"),
					publication: ploc("publicationDescription"),
					statement: ploc("statementDescription"),
				},
				citationTypes: {
					hearsay: ploc("hearsay"),
					publication: ploc("publication"),
					statement: ploc("statement"),
				},
				submit: ploc("submit"),
			},
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<AboutPage />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
