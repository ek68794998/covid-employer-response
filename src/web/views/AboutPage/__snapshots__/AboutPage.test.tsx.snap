// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> renders without exploding 1`] = `
<main
  id="about"
>
  <div
    className="AboutPage__Container"
  >
    <div
      className="AboutPage__Content"
    >
      <h1>
        Fřè9úèñƭℓ¥ áƨƙèδ 9úèƨƭïôñƨ
      </h1>
      <h2
        id="what-is"
      >
        Wλáƭ ïƨ Çè₥ƥřèƨ?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(**Çè₥ƥřèƨ** (ÇôřôñáƲïřúƨ É₥ƥℓô¥èř Rèƨƥôñƨè Třáçƙèř) ïƨ á çôℓℓáβôřáƭïƲè, ôƥèñ-ƨôúřçè ƥřôJèçƭ δèƨïϱñèδ ƭô ƭřáçƙ ƭλè řèƨƥôñƨèƨ ôƒ è₥ƥℓô¥èřƨ ƭô ƭλè [çôřôñáƲïřúƨ](https://en.wikipedia.org/wiki/Coronavirus_disease_2019) (ÇÓVÌÐ-19) [ƥáñδè₥ïç ôƒ 2019&ñδáƨλ;2020](https://en.wikipedia.org/wiki/2019%E2%80%9320_coronavirus_pandemic). Ìƭ ïƨ βúïℓƭ úƨïñϱ ïñƒôř₥áƭïôñ ƨôúřçèδ ƒřô₥ á ñú₥βèř ôƒ ƥℓáçèƨ áñδ çáƭèϱôřïƺèƨ ïƭ βáƨèδ ôñ λôω řèℓïáβℓè ƭλè ôřïϱïñáℓ ïñƒôř₥áƭïôñ ƨôúřçè ïƨ çôñƨïδèřèδ ƭô βè. Fôř ₥ôřè ïñƒôř₥áƭïôñ, ƥℓèáƨè Ʋïƨïƭ [ƭλè ƥřôJèçƭ'ƨ ƥáϱè ôñ GïƭHúβ](https://github.com/ekumlin/covid-employer-response).) /]
      <h2
        id="contributing"
      >
        Çáñ Ì λèℓƥ?
      </h2>
      [component: ReactMarkdown source=(Ýèƨ! Çè₥ƥřèƨ ïƨ ôƥèñ-ƨôúřçè áñδ ôƥèñ-δáƭá áñδ ωè'δ ℓôƲè ƭô λáƲè ¥ôúř λèℓƥ! Ýôú çáñ λèℓƥ β¥ ƨúβ₥ïƭƭïñϱ ñèω è₥ƥℓô¥èř çℓáï₥ƨ áñδ çλáñϱèƨ, ôř β¥ ƨúβ₥ïƭƭïñϱ çôδè çλáñϱèƨ ƭô ƭλè ƨïƭè. §èè [ôúř ƨúβ₥ïƨƨïôñƨ ƥáϱè](/ƨúβ₥ïƭ) ƒôř ₥ôřè.) /]
      <h2
        id="reports-and-claims"
      >
        Wλáƭ ářè è₥ƥℓô¥èř çℓáï₥ƨ áñδ řèƥôřƭƨ?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(Âñ **è₥ƥℓô¥èř çℓáï₥** ïƨ ôñè ôř ₥ôřè δèçℓářáƭïôñƨ áβôúƭ áñ è₥ƥℓô¥èř ωλïçλ ïñçℓúδè á ƨú₥₥ář¥ áñδ çïƭáƭïôñƨ ƭô βáçƙ ïƭ úƥ. Tλïƨ áƥƥèářƨ áƨ á ƨïñϱℓè βúℓℓèƭ-ƥôïñƭ ïñ ƭλè è₥ƥℓô¥èř'ƨ ℓïƨƭïñϱ.

Âñ **è₥ƥℓô¥èř řèƥôřƭ** ïƨ á çôℓℓèçƭïôñ ôƒ çℓáï₥ƨ áβôúƭ áñ è₥ƥℓô¥èř áƨ ωèℓℓ áƨ áñ ôƲèřáℓℓ řáƭïñϱ ôƒ λôω ωèℓℓ ƭλáƭ è₥ƥℓô¥èř λáƨ λáñδℓèδ ƭλè ÇÓVÌÐ-19 ƥáñδè₥ïç çřïƨïƨ.
) /]
      <h2
        id="claim-processing"
      >
        Hôω δô çℓáï₥ƨ ϱèƭ ƥřôçèƨƨèδ?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(Óñçè á çℓáï₥ ïƨ ƨúβ₥ïƭƭèδ, ïƭ ₥úƨƭ βè řèƲïèωèδ β¥ á çôñƭřïβúƭôř ƭô ƭλè ƥřôJèçƭ βèƒôřè ïƭ ωïℓℓ βè áƥƥřôƲèδ áñδ áƥƥèář ôñ ƭλè ƨïƭè. Tλïƨ řèƲïèω ƥřôçèƨƨ ωïℓℓ řè9úïřè ƭλè çôñƭřïβúƭôř ƭô ϱô ƭλřôúϱλ èáçλ ôƒ ƭλè ƨôúřçèƨ áñδ Ʋáℓïδáƭè ƭλáƭ ƭλè çℓáï₥ ₥áδè ïƨ ƨúβƨƭáñƭïáƭèδ β¥ ƭλè ƨôúřçè. Óñçè ƭλè ƨúβ₥ïƭƭèř áñδ çôñƭřïβúƭôř(ƨ) áϱřèè ôñ ƭλè ƨúβ₥ïƭƭèδ ïñƒôř₥áƭïôñ, ƭλè çℓáï₥ ïƨ çô₥ƥℓèƭèδ áñδ ωïℓℓ áƥƥèář ƭλè ñèжƭ ƭï₥è ƭλè ƨïƭè ïƨ úƥδáƭèδ.) /]
      <h2
        id="submit-claims"
      >
        Hôω δô Ì áδδ, çλáñϱè, ôř δïƨƥúƭè çℓáï₥ƨ?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(Tô ƨúβ₥ïƭ á çℓáï₥ çλáñϱè ôƒ áñ¥ ƙïñδ, ƥℓèáƨè çℓïçƙ ƭλè "§úβ₥ïƭ" ℓïñƙ ïñ ƭλè ñáƲïϱáƭïôñ ₥èñú. Fřô₥ ƭλèřè, ôú çáñ èïƭλèř ƨúβ₥ïƭ á řè9úèƨƭ ƒôř ƨô₥èôñè ƭô úƥδáƭè ôúř δáƭá ƒïℓèƨ, ôř ƨúβ₥ïƭ á GïƭHúβ ƥúℓℓ řè9úèƨƭ ƭô úƥδáƭè ƭλè ƒïℓèƨ ¥ôúřƨèℓƒ.) /]
      <h2
        id="citation-types"
      >
        Wλáƭ δôèƨ ƭλè ïçôñ ñèжƭ ƭô á çℓáï₥ ₥èáñ?
      </h2>
      [component: ReactMarkdown className=(AboutPage__CitationTypes) linkTarget=(_blank) source=(- Þúβℓïçáƭïôñ: Þúβℓïƨλèδ ářƭïçℓèƨ ƨúçλ áƨ á çô₥ƥáñ¥ βℓôϱ, ÞR ƨƭáƭè₥èñƭ, ϱôƲèřñ₥èñƭ řèƥôřƭ, ôř ℓèáƙèδ δôçú₥èñƭ.
- §ƭáƭè₥èñƭ: Þúβℓïƨλèδ ƨƭáƭè₥èñƭƨ ƒřô₥ è₥ƥℓô¥èèƨ (ƒôř₥èř, áñôñ¥₥ôúƨ, ôř ôƭλèřωïƨè).
- Hèářƨá¥: Wôřδ-ôƒ-₥ôúƭλ, řú₥ôřƨ, ôř úñçôñƒïř₥èδ ïñƒôř₥áƭïôñ.
) /]
      <h2
        id="employer-ratings"
      >
        Wλáƭ δèƭèř₥ïñèƨ ƭλè řáƭïñϱ ôƒ áñ è₥ƥℓô¥èř?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(Ráƭïñϱƨ ářè áℓϱôřïƭλ₥ïçáℓℓ¥ δèƭèř₥ïñèδ βáƨèδ ôñ ƭλè ƥôƨïƭïƲïƭ¥ áñδ ñèϱáƭïƲïƭ¥ ôƒ ƭλè çℓáï₥ƨ ƭλáƭ ïƭ λáƨ. Çℓáï₥ƨ ƭλáƭ ářè ₥ôřè ƨáƒèƭ¥-çřïƭïçáℓ λáƲè á λïϱλèř ωèïϱλƭ ƭλáñ ƭλôƨè ωλïçλ ářè ñôƭ, áñδ ƭλúƨ ářè ñôƭ ƒáçƭôřèδ áƨ λèáƲïℓ¥.) /]
      <h2
        id="layoffs-and-furloughs"
      >
        Wλáƭ ïƨ ƭλè δïƒƒèřèñçè βèƭωèèñ á "ℓá¥ôƒƒ" áñδ á "ƒúřℓôúϱλ"?
      </h2>
      [component: ReactMarkdown linkTarget=(_blank) source=(Fôř ƭλè ƥúřƥôƨèƨ ôƒ ƭλïƨ ƨïƭè, á **ƒúřℓôúϱλ** ïñδïçáƭèƨ ƭλáƭ áƒƒèçƭèδ è₥ƥℓô¥èèƨ ářè ñôƭ ωôřƙïñϱ, βúƭ ářè ƨƭïℓℓ ₥ářƙèδ áƨ è₥ƥℓô¥èδ β¥ ƭλè çô₥ƥáñ¥. Tλè¥ ₥á¥ řèçèïƲè ƥářƭïáℓ ƥá¥ ôř βèñèƒïƭƨ, áñδ ářè ϱèñèřáℓℓ¥ èжƥèçƭèδ ƭô βè řèλïřèδ ïñ ƭλè ƒúƭúřè. Â **ℓá¥ôƒƒ** ïñδïçáƭèƨ ƭλáƭ áƒƒèçƭèδ è₥ƥℓô¥èèƨ λáƲè βèèñ ƭèř₥ïñáƭèδ ƒřô₥ ƭλèïř Jôβƨ èñƭïřèℓ¥. Â ₥ôřè çô₥ƥℓèƭè δèƨçřïƥƭïôñ ïƨ áƲáïℓáβℓè [ôñ Ðïçƭïôñář¥.çô₥](λƭƭƥƨ://ωωω.δïçƭïôñář¥.çô₥/è/ƒúřℓôúϱλ-Ʋƨ-ℓá¥ôƒƒ/).

§ô₥è δïáℓèçƭƨ ôƒ Éñϱℓïƨλ úƨè ƭλè ƭèř₥ "ƭè₥ƥôřář¥ ℓá¥ôƒƒ" ƭô δèƨçřïβè á ƒúřℓôúϱλ, βúƭ ƭô áƲôïδ çôñƒúƨïôñ, ωè áƲôïδ ƭλáƭ ƭèř₥. Âƨ á řèƨúℓƭ, ïñƒôř₥áƭïôñ ωè çïƭè ƒřô₥ ƨô₥è ƨôúřçèƨ ₥á¥ βè èřřôñèôúƨℓ¥ ωřïƭƭèñ áƨ "ℓá¥ôƒƒ" ωλèñ ƭλè¥ ářè ïñƨƭèáδ řèƒèřřïñϱ ƭô á ƒúřℓôúϱλ. §λôúℓδ ¥ôú èƲèř řúñ ïñƭô á ƨïƭúáƭïôñ ℓïƙè ƭλïƨ, ƥℓèáƨè [ƨúβ₥ïƭ á çôřřèçƭïôñ](/ƨúβ₥ïƭ)!
) /]
    </div>
  </div>
</main>
`;
